### PR TITLE
Assignment#1 20기 분석 이시내

### PR DESCRIPTION
**Pull request by joljol38**

### CODE REVIEW

**Comment by okaybody10** [comment]
이거 A1, A2까지 한거 맞으실까요??
그냥 정말 말 그대로 궁금해서 여쭤보는겁니당

**Comment by joljol38** [comment]
네...! 혹시 뭐 잘못되었을까요..?

**Comment by okaybody10** [comment]
> 네...! 혹시 뭐 잘못되었을까요..?

아뇨 그냥 진짜 말 그대로 궁금해서 여쭤봤어요! 별 의미없어요
고생하셨어요

File: softmax.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    #############################################################################
+    # TODO: Compute the softmax loss and its gradient using no explicit loops.  #
+    # Store the loss in loss and the gradient in dW. If you are not careful     #
+    # here, it is easy to run into numeric instability. Don't forget the        #
+    # regularization!                                                           #
+    #############################################################################
+    # *****START OF YOUR CODE (DO NOT DELETE/MODIFY THIS LINE)*****
+
+    score = X.dot(W)
+    score -= np.max(score)
```
Comment (okaybody10) [inline softmax.py:81]: 나중에 합계를 구할 때 가로를 움직이면서 구할 것이기 때문에, 최댓값도 여기에서 구해주시면 됩니다.
즉, ```np.max(score, axis=1).reshape(-1, 1)``` 혹은 ```np.amax(score, axis=1, keepdims=True)```로 구현하시면 좋을 것 같네요

File: softmax.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    dS = score_exp / score_exp_sum.reshape(-1, 1)
+    dS[range(X.shape[0]), y] -= 1
+
+    dW = (X.T).dot(dS)
+
+    loss /= X.shape[0]
+    dW /= X.shape[0]
+
+    loss += 0.5 * reg * np.sum(W * W)
```
Comment (okaybody10) [inline softmax.py:97]: Regularization term에 대한 loss를 구할 때 0.5를 곱해주시면 안됩니다.
즉, ```loss += reg * np.sum(W*W)```로 바꿔주시면 됩니다.

[update] 라기보다는 앞쪽에 미리 반영을 해줬네요 (87번째 줄에 Reg term이 있네요)
그렇기 떄문에, 94, 97번째줄 둘 다 제거해주셔야 합니다.

**Review by okaybody10 — COMMENTED** [review]
Code Review I End
